### PR TITLE
Lua: provide global `PANDOC_WRITER_OPTIONS` [API change]

### PR DESCRIPTION
**Pull request by tarleb**
Closes: #5221

### CODE REVIEW

**Comment by tarleb** [comment]
The template and syntax map are not exposed, as there is no JSON instance for them and there is no clear benefit of making them available.

File: src/Text/Pandoc/Lua/Global.hs [hunk 1]
```diff
@@ -58,8 +60,11 @@ setGlobal global = case global of
   PANDOC_READER_OPTIONS ropts -> do
     pushReaderOptionsReadonly ropts
     Lua.setglobal "PANDOC_READER_OPTIONS"
+  PANDOC_WRITER_OPTIONS ropts -> do
```
Comment (jgm) [inline src/Text/Pandoc/Lua/Global.hs:63]: Could be less confusing to use `wopts` here.

**Comment by jgm** [comment]
Question: would it make sense to use pushViaJSON for the ReaderOptions as well?  This might help keep things in sync in the future, if ReaderOptions change, with less manual intervention.

**Comment by tarleb** [comment]
My one argument against `pushReaderOptions = pushViaJSON` would be that JSON uses dashes in field names, which are a less nice to use in Lua: `reader.strip_comments = true` vs `reader['strip-comments'] = true`.  But for many of the fields, going through JSON would give a much nicer and consistent interface. I'll hack up a PR for that.

**Comment by jgm** [comment]
That's true.  On the other hand, if it's uniform between JSON and Lua, that's less to remember.
One possibility would be to have pushViaJSON systematically change `-` to `_` in field names, but maybe that's not worth it.

**Comment by tarleb** [comment]
I hadn't considered the possibility of replacing `-` with `_`, that seems nice. The only interesting differences would then be

- `indented_code_classes` would no longer be a `List` (but I want to fix that with the next release of hslua-aeson), and
- `abbreviations` are currently a table with the abbreviations as keys and `true` as values, which is the idiomatic way to represent sets in Lua; it would become a simple List when going through JSON.

I don't believe many people would be impacted by either of these.

**Comment by jgm** [comment]
I guess the "set" representation of abbreviations would be more efficient if it were actually used in a lua filter.  But it probably doesn't matter much.  After all, if efficiency were an issue, the filter could always construct the "set" like table from the list like one.

**Comment by tarleb** [comment]
I think I want to wait for the next hslua-aeson release before doing this, as the released version as a few issues that might get in the way. But there is e58a5ceed8bb629efbfd670bf162b65b7853ab7f as a first step into that direction.

**Comment by jgm** [comment]
Sounds good.  By "doing this", I assume you mean changing `PANDOC_READER_OPTIONS` to use via json.  Is this PR for `PANDOC_WRITER_OPTIONS` ready to merge?

**Comment by tarleb** [comment]
The code should be ok, but I just noticed that the API change isn't really described in the commit message; I will fix that.

**Comment by tarleb** [comment]
Better now, I think.

**Pull request merged**